src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,160 @@
+        System.out.println("Saved successfully!");
+    }
+
+    /**
+     * Shows the tasks in taskList
+     * @param taskList the tasks to be shown
+     * @return string that contains tasks in list
+     */
+    public static String showTasksInList(TaskList taskList) {
+        StringBuilder stringBuilder = new StringBuilder("Here are the tasks in your list:\n");
+        for (int i = 0; i < taskList.taskListSize(); i++) {
+            stringBuilder.append(i + 1 + ". " + taskList.getTask(i).toString() + "\n");
+        }
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Reads the command entered by user
+     * @return the command entered by user
+     */
+    public String readCommand() {
+        return sc.nextLine();
+    }
+
+    /**
+     * Shows the error message
+     * @param message the error message
+     * @return string that contains error message
+     */
+    public String showError(String message) {
+        return message;
+    }
+
+    /**
+     * Shows when task is added
+     * @param taskList the taskList that the task is added to
+     * @return string that contains task added
+     */
+    public String showTaskAdded(TaskList taskList) {
+        return ("Got it. I've added this task:")
+                + "\n" + (taskList.latestTask() + "\nNow you have " + taskList.taskListSize()
+                + " tasks in the list.");
+    }
+
+    /**
+     * Shows when task is deleted
+     * @param taskList the taskList that the task is deleted from
+     * @param deletedTask the task which is deleted
+     * @return string that shows task deleted
+     */
+    public String showTaskDeleted(TaskList taskList, Task deletedTask) {
+        return ("Noted. I've removed this task:\n"
+                + deletedTask) +
+                ("\nNow you have " + (taskList.taskListSize()) + " tasks in the list.");
+    }
+
+    /**
+     * Shows when task is marked as done
+     * @param taskList the taskList that the task is marked from
+     * @param integer the index of the task which is marked
+     * @return string that shows task marked as done
+     */
+    public String showMarkAsDone(TaskList taskList, int integer) {
+        return ("Nice! I've marked this task as done:\n"
+                + taskList.getTask(integer - 1).toString());
+    }
+
+    /**
+     * Shows when task is marked as not done
+     * @param taskList the taskList that the task is marked from
+     * @param integer the index of the task which is marked as not done
+     * @return string that shows task marked as not done
+     */
+    public String showMarkAsNotDone(TaskList taskList, int integer) {
+        return ("OK, I've marked this task as not done yet:\n"
+                + taskList.getTask(integer - 1).toString());
+    }
+
+    /**
+     * Displays Ui when task is found
+     * @param taskList the tasklist to be displayed
+     * @return string that contains tasks found
+     */
+    public String showTaskFound(TaskList taskList) {
```

Comment:
Since there might be more than 1 task that are matched, perhaps we could rename this method to showTasksFound